### ISSUE TITLE
Algoryn Validation Shows Strategic Use Incorrectly

### ISSUE THREAD

**Issue by fungus1487** [BSData/gates-of-antares#117]
Taken from the Facebook Group:

> Hi, I just discovered a bug in the Algoryn list. It appears to think it has 3 units in strategic even though I added none. when i add one it goes to 4.

![image](https://cloud.githubusercontent.com/assets/1430843/25339883/f0becf14-28fb-11e7-9cd0-12979f9888ef.png)

**Comment by rythos42** [comment]
This is by-design, but there are other design choices we can make.

To start -- the user that reported this just hasn't entered a roster size. They left it at -1, which is "infinite", which is not a valid Gates of Antares roster size.

When I implemented this behavior I had a choice to make about what to do about this, and in the end I choose the implementation that was the least amount of work for me. :P There are 3 options (that I can see):

1. (existing) User must enter a roster size, and validation happens. If they don't enter a roster size, the validation is meaningless because their roster size is meaningless. "Infinite" is not a supported size - every roster **must** be a size in the game, because unlike in 40k the size of your game changes your force org. 

1a. The previous implementation was very similar, except that you _couldn't make a roster without entering a size_. This was inconvenient, because you had to pick from a massive drop-down list. I feel the new implementation is easier for the user, because the drop-down is smaller. The trade-off is that entering the roster size can be missed, as this user did.

2. No validation if the user doesn't enter a valid roster size. This is a reasonable method of solving the problem, but it was more work (that I didn't want to do) and because "infinite" is invalid I thought this was a "garbage in, garbage out" situation. Just enter a valid roster size and you'll get real information. BS data doesn't have a "check for infinite size roster", last I checked, so it got messy quickly as far as implementing it goes.

3. Have the validation be based on the _currently entered points value_, rather than the "max roster size". On the face of it, this seems like a good idea, but it could get weird. The user will see validation telling them to enter 3 Tactical choices, then they'll enter that 3rd Tactical choice and suddenly the validation could change to say enter 4 Tactical Choices (because they crossed a roster size boundary, from 750 to 1000, for example). It's a constantly moving target, until they settle on a size.

This is just an explanation of what and why - I don't have time to change anything right now, but if someone has a comment on what they feel should be done I'd be interested in talking more about it.

Oh, and this is for every roster, not just Algoryn.

**Comment by fungus1487** [comment]
Personally I think 1a is the best option, it forces the user to choose the roster size.

From a game perspective this makes complete sense:

"hey fred, do you want a game of antares?"

"sure bob, 1000pts?"

"let's make it 1,250pts?"

"OK, see you Wednesday"

**Comment by rythos42** [comment]
Feel free to revert it back if you really like it. I found that drop down list annoying to use, so I changed it. It was a list of ~40 options, only 5 of which would ever applied to you. It was unsorted (which is a different issue) so it was a pain to find the one you wanted, even after you scrolled down to "Concord". 

Your dialog applies to the current implementation too -- it's just the user has to enter the roster size in an input field instead of a drop down list.
